packages/dynamoose/lib/Model/index.ts

```diff
@@ -443,7 +445,7 @@ export class Model<T extends ItemCarrier = AnyItem> extends InternalPropertiesCl
 			const params: DynamoDB.BatchGetItemInput = {
 				"RequestItems": {
 					[table.getInternalProperties(internalProperties).name]: {
-						"Keys": await Promise.all(keyObjects.map(async (key) => this.Item.objectToDynamo(await this.Item.objectFromSchema(key, this, {"type": "toDynamo", "modifiers": ["set"], "typeCheck": false, "mapAttributes": true}))))
+						"Keys": await Promise.all((await Promise.all(keyObjects)).map(async (key) => this.Item.objectToDynamo(await this.Item.objectFromSchema(key, this, {"type": "toDynamo", "modifiers": ["set"], "typeCheck": false, "mapAttributes": true}))))
```

Comment:
Hmm. I think this is more complicated than it needs to be. There is no reason we should need to convert `convertObjectToKey` to an async function.

Additionally, I'm really confused about why this code doesn't work in the first place. With your change we are calling `objectFromSchema` 2 times. Once in the `convertObjectToKey` & another on this line. Even this time in the previous code, we had `"mapAttributes": true`. So this **should** work.

I don't like the idea of calling `objectFromSchema` twice in this function. That seems highly inefficient.

I'd like to understand more about why it's not mapping the attributes on this line (before this PR) since we have `"mapAttributes": true`.

If you'd like to dig into that, that'd be awesome. Otherwise I'll try to take a look at that at some point.